sql/src/main/java/io/crate/metadata/Functions.java

```diff
@@ -54,19 +54,21 @@ private void generateFunctionResolvers(Map<FunctionIdent, FunctionImplementation
     }
 
     /**
-     * Returns the function implementation for the given function name and arguments.
+     * Returns the function implementation for the given function schema, name and arguments.
+     * <p>
      *
+     * @param schema         The schema name.
      * @param name           The function name.
      * @param argumentsTypes The function argument types.
-     * @return The function implementation..
+     * @return a function implementation
      * @throws UnsupportedOperationException if an implementation is not found.
      */
-    public FunctionImplementation getSafe(String name, List<DataType> argumentsTypes)
+    public FunctionImplementation getSafe(@Nullable String schema, String name, List<DataType> argumentsTypes)
```

Comment:
I think it would be easier on a API user if this is two methods. E.g. `getBuiltin` / `getUDF`, this would clarify how the API has to be used (e.g. builtin first, then as fallback UDF).

Otherwise it would be tempting to always pass in the default/session schema in the first call